test/net/sourceforge/kolmafia/objectpool/ConcoctionTest.java

```diff
@@ -30,61 +52,51 @@ public void itShouldSortUsables() {
     assertEquals(usableList.size(), thing);
   }
 
-  // tests the portion of the contract that says sgn(x.compareTo(y)) == -sgn(y.compareTo(x) and
-  // (x.compareTo(y)==0) == (x.equals(y))
+  // sgn(x.compareTo(y)) == -sgn(y.compareTo(x)
   @Test
-  @Disabled("Nested for-loops are slow...")
-  public void itShouldBeSymmetric() {
-    LockableListModel<Concoction> first = ConcoctionDatabase.getUsables();
-    LockableListModel<Concoction> second = ConcoctionDatabase.getUsables();
-    for (Concoction acon : first) {
-      for (Concoction bcon : second) {
-        int x = acon.compareTo(bcon);
-        int y = bcon.compareTo(acon);
-        String msg = acon.toString() + " * " + bcon.toString();
-        assertEquals(sgn(x), -sgn(y), msg);
-        if (x == 0) assertEquals(acon, bcon, msg);
+  public void compareToShouldBeQuasiSymmetric() {
+    for (int i = 0; i < maxIndex; i++) {
+      for (int j = 0; j < maxIndex; j++) {
+        msg = "comparing (quasi symmetry)" + idsAndMaps.get(i) + "and " + idsAndMaps.get(j);
+        assertEquals(sgn(result[i][j]), -sgn(result[j][i]), msg);
```

Comment:
sgn is redundant here, since you've already invoked it in populating this array.